src/vmspawn/vmspawn.c

```diff
@@ -731,20 +732,17 @@ static int run_virtual_machine(void) {
                 log_warning("Couldn't find OVMF firmware blob with Secure Boot support, "
                             "falling back to OVMF firmware blobs without Secure Boot support.");
 
-        const char *accel = use_kvm ? "kvm" : "tcg";
-        if (IN_SET(native_architecture(), ARCHITECTURE_ARM64, ARCHITECTURE_ARM64_BE))
-                machine = strjoin("type=virt,accel=", accel);
-        else
-                machine = strjoin("type=q35,accel=", accel, ",smm=", on_off(ovmf_config->supports_sb));
-        if (!machine)
-                return log_oom();
-
         r = find_qemu_binary(&qemu_binary);
         if (r == -EOPNOTSUPP)
                 return log_error_errno(r, "Native architecture is not supported by qemu.");
         if (r < 0)
                 return log_error_errno(r, "Failed to find QEMU binary: %m");
 
+        if (IN_SET(native_architecture(), ARCHITECTURE_ARM64, ARCHITECTURE_ARM64_BE))
+                machine = "type=virt";
+        else
```

Comment:
We should only apply this on x86 and fail for other architectures as we don't know the machine to use in that case

Reply:
https://github.com/systemd/mkosi/blob/a503c80dca9c7f569d9333f89f85d3f5ada60859/mkosi/config.py#L368-L384
Would it not be more correct to implement something like this from mkosi?